src/FakeItEasy/Expressions/ExpressionArgumentConstraintFactory.cs

```diff
@@ -77,7 +80,7 @@ private static object InvokeExpression(Expression expression)
             return Expression.Lambda(expression).Compile().DynamicInvoke();
         }
 
-        private IArgumentConstraint GetArgumentConstraintFromExpression(Expression expression)
+        private IArgumentConstraint GetArgumentConstraintFromExpression(Expression expression, out object value)

```

Comment:
`out`s always look funny to me. It's a private method, so not a huge deal.
Consider returning a compound object?
If you still prefer the `out`, I will hit the "merge" button.